### PR TITLE
Optimizations on contracts

### LINKED ISSUE #1114: Contract optimizations
Implement: 

```
ArbitrumL1Messenger, line 12: Note, the Resolver import could be removed.
ArbitrumL1Messenger, line 76: Optimization, in the sendMessage() function the deposits[tx.origin] value is read from storage twice.

FillManager, line 106: Optimization, in the fillRequest() function the l1Resolver variable is read from storage twice.
FillManager, line 113. Note, the fillRequest() function updates fills[requestId] in storage after doing external calls. It is a general recommendation to apply effects before external interactions.
FillManager, line 146: Optimization, in the invalidateFill() function the l1Resolver variable is read from storage twice.

RequestManager, line 37. Optimization, the Request.validUntil field could fit into uint32.
RequestManager, line 38. Optimization, the Request.lpFee field could fit into uint88.
RequestManager, line 39. Optimization, the Request.protocolFee field could fit into uint88.
RequestManager, line 40. Optimization, the Request.activeClaims field could fit into a smaller type.
RequestManager, line 41. Optimization, the Request.activeClaims field could fit into a smaller type.
RequestManager, line 288. Optimization, the createRequest() function reads currentNonce from storage thrice. Consider introducing a local variable.
RequestManager, line 315. Optimization, the createRequest() function excessively reads newRequest.validUntil from storage instead of having it in a local variable or calculate inplace.
RequestManager, line 376. Optimization, the claimRequest() function reads currentNonce from storage four times. Consider introducing a local variable.
RequestManager, line 381. Optimization, the claimRequest() function reads claimStake from storage twice. Consider introducing a local variable or use msg.value.
RequestManager, line 382. Optimization, the claimRequest() function excessively assigns address(0) to claim.lastChallenger. All values in storage are 0 by default.
RequestManager, line 383. Optimization, the claimRequest() function excessively assigns 0 to claim.challengerStakeTotal. All values in storage are 0 by default.
RequestManager, line 384. Optimization, the claimRequest() function excessively assigns 0 to claim.withdrawnAmount. All values in storage are 0 by default.
RequestManager, line 434. Optimization, the challengeClaim() function reads claims[claimId] properties from storage multiple times.
RequestManager, line 456. Optimization, in the challengeClaim() function the msg.sender == nextActor requirement could be put in the above else clause and replaced with masg.sender == claim.claimer. After this change you do not need the nextActor variable anymore and should use msg.sender instead.
RequestManager, line 460: Optimization, the challengeClaim() function reads claim.claimerStake from storage twice.
RequestManager, line 462. Optimization, the challengeClaim() could pointlessly update claim.lastChallenger with it's own value.
RequestManager, line 464. Optimization, the challengeClaim() function reads the claim.challengerStakeTotal from storage twice.
RequestManager, line 467. Optimization, the challengeClaim() could pointlessly update claim.termination with it's own value.
RequestManager, line 472. Optimization, in the challengeClaim() function the requirement claim.termination >= minimumTermination is always true.
RequestManager, line 473. Optimization, the challengeClaim() reads claim.claimTermination from storage multiple times.
RequestManager, line 504: Optimization, the withdraw() function reads claims[claimId] properties from storage multiple times.
RequestManager, line 515. Optimization, in the withdraw() function the claim.withdrawnAmount variable is read from stoage thrice.

Resolver, line 60. Optimization, in the resolve() function the sourceCHainInfo fields read twice from storage. Read it into memory to save gas.
```

### PR DESCRIPTION
**Pull request by fredo**
With the prereport of the audit some optimizations were suggested.

closes #1114

### CODE REVIEW

**Review by czepluch — APPROVED** [review]
lgtm

File: contracts/contracts/RequestManager.sol [hunk 1]
```diff
@@ -276,33 +276,33 @@ contract RequestManager is Ownable, LpWhitelist, RestrictedCalls {
 
         uint256 lpFeeTokenAmount = lpFee(amount);
         uint256 protocolFeeTokenAmount = protocolFee(amount);
-        uint256 totalTokenAmount = amount +
```
Comment (GabrielBuragev) [inline contracts/contracts/RequestManager.sol:279]: Out of curiosity, what was the reason for removing the `totalTokenAmount` here? Is that also a part of the optimizations?

**Comment by GabrielBuragev** [comment]
This PR covers all the requirements.:heavy_check_mark:  Nicely done!

File: contracts/contracts/RequestManager.sol (hunk 1 above)
Comment (czepluch) [reply contracts/contracts/RequestManager.sol:279]: I think the idea is that we don't use a storage slot, and instead just calculate it directly based on values that we already have.

Comment (GabrielBuragev) [reply contracts/contracts/RequestManager.sol:279]: Aha, now it makes sense to me. Thanks!

Comment (fredo) [reply contracts/contracts/RequestManager.sol:279]: A function only has 16 slots on the stack (for parameters and local variables). In this function we reached the maximum amount but read storage variables multiple times. So in order to copy storage values into memory and save costs for reading from storage I had to free up some variables, like `totalTokenAmount`

Comment (GabrielBuragev) [reply contracts/contracts/RequestManager.sol:279]: Great explainer! thanks.

**Comment by GabrielBuragev** [comment]
:heavy_check_mark:

**Pull request merged**